### PR TITLE
Sample Update

### PR DESCRIPTION
**Pull request by MatthewDittrich**
Changes:
-adding Run 3 'other' samples
-new Run 3 skims including needed branches for JERC
-New Run 2 skims including needed branches for JERC

### CODE REVIEW

**Comment by MatthewDittrich** [comment]
Run 2 Percent Change:
Screen Shot 2024-08-24 at 4 30 30 PM


Ref Yield Tables: [Run2_Ref_Tables/](http://uaf-10.t2.ucsd.edu/~mdittric/hpg/Run2_Ref_Tables/)
New Yield Tables: [Run2_New_Tables/](http://uaf-10.t2.ucsd.edu/~mdittric/hpg/Run2_New_Tables/)

**Comment by MatthewDittrich** [comment]
Run 3 Percent Change: 
Screen Shot 2024-08-24 at 4 29 57 PM

**Comment by MatthewDittrich** [comment]
As of now, the skims are not merged (I have not been able to successfully merge them without errors or yields changing significantly)

File: analysis/wwz/get_wwz_yields.py [hunk 1]
```diff
@@ -719,7 +719,7 @@ def make_plots(histo_dict,grouping_mc,grouping_data,save_dir_path,apply_nsf_to_c
             #if "bdt" in cat_name: continue
             #if cat_name not in ["sr_4l_sf_incl", "sr_4l_of_incl", "cr_4l_btag_of", "cr_4l_btag_sf_offZ_met80", "cr_4l_sf", "sr_4l_bdt_sf_trn", "sr_4l_bdt_of_trn"]: continue # TMP
             #if cat_name not in ["cr_4l_sf_higgs"]: continue
-            if cat_name not in ["cr_4l_btag_of", "cr_4l_btag_sf_offZ_met80", "cr_4l_sf"]: continue
+            #if cat_name not in ["cr_4l_btag_of", "cr_4l_btag_sf_offZ_met80", "cr_4l_sf"]: continue
```
Comment (kmohrman) [inline analysis/wwz/get_wwz_yields.py:722]: Let's leave this uncommented by default (usually we don't want to make plots for _everything_)

File: analysis/wwz/get_wwz_yields.py [hunk 2]
```diff
@@ -935,9 +935,9 @@ def main():
 
     # Make plots
     if args.make_plots:
-        make_plots(histo_dict,sample_dict_mc,sample_dict_data,save_dir_path=out_path,apply_nsf_to_cr=False)
```
Comment (kmohrman) [inline analysis/wwz/get_wwz_yields.py:938]: Let's leave the standard plotting script the one tunred on by default

File: analysis/wwz/make_datacards.py [hunk 3]
```diff
@@ -32,6 +32,20 @@
...
+    "btagSFbc_uncorrelated_2016APV"    : {"yr_rel":"UL16APV", "yr_notrel": ["UL16", "UL17", "UL18", "2022", "2022EE"]},
+    "btagSFlight_uncorrelated_2016"    : {"yr_rel":"UL16", "yr_notrel": ["UL16APV", "UL17", "UL18", "2022", "2022EE"]},
+    "btagSFbc_uncorrelated_2016"       : {"yr_rel":"UL16", "yr_notrel": ["UL16APV", "UL17", "UL18", "2022", "2022EE"]},
+    "btagSFlight_uncorrelated_2017"    : {"yr_rel":"UL17", "yr_notrel": ["UL16APV", "UL16", "UL18", "2022", "2022EE"]},
+    "btagSFbc_uncorrelated_2017"       : {"yr_rel":"UL17", "yr_notrel": ["UL16APV", "UL16", "UL18", "2022", "2022EE"]},
+    "btagSFlight_uncorrelated_2018"    : {"yr_rel":"UL18", "yr_notrel": ["UL16APV", "UL16", "UL17", "2022", "2022EE"]},
+    "btagSFbc_uncorrelated_2018"       : {"yr_rel":"UL18", "yr_notrel": ["UL16APV", "UL16", "UL17", "2022", "2022EE"]},
+
+    "btagSFbc_uncorrelated_2022"       : {"yr_rel":"2022", "yr_notrel": ["UL16APV","UL16","UL17","UL18","2022EE"]},
+    "btagSFbc_uncorrelated_2022EE"     : {"yr_rel":"2022EE", "yr_notrel": ["UL16APV","2022"]},
```
Comment (kmohrman) [inline analysis/wwz/make_datacards.py:46]: What about the rest of R2?

File: analysis/wwz/make_datacards.py [hunk 4]
```diff
@@ -220,6 +236,22 @@ def get_rate_systs(proc_lst):
...
+def determine_updo_same(nom,up,down):
+    if nom < 0:
+        raise Exception("Negative values should have been fixed by this point!")
+    elif ((up > nom) and (down > nom)):
+        return True
+    elif ((up < nom) and (down < nom)):
+        return True
+    else: return False
+
+def fix_updown_same(nom,up,down):
```
Comment (kmohrman) [inline analysis/wwz/make_datacards.py:248]: Add a comment before this function that explains what it is and what it does

File: analysis/wwz/make_datacards.py [hunk 5]
```diff
@@ -251,7 +283,8 @@ def get_syst_base_name_lst(in_lst):
                 valvar_kappa_do = yt.valvar_op(valvar_do,valvar_nom,"div")
 
                 # Handle negative cases
-                if (valvar_kappa_up[0]<=0) and (valvar_kappa_do[0]<=0): raise Exception("Both kappas negative, should not be possible.")
+                if (valvar_kappa_up[0]<=0) and (valvar_kappa_do[0]<=0):
```
Comment (kmohrman) [inline analysis/wwz/make_datacards.py:286]: This seems to only check the negative direction?

File: analysis/wwz/make_datacards.py [hunk 6]
```diff
@@ -371,7 +404,7 @@ def main():
     parser.add_argument("--do-tf",action="store_true",help="Do the TF data-driven background estimation")
     parser.add_argument("--bdt",action="store_true",help="Use BDT SR bins")
     parser.add_argument("--unblind",action="store_true",help="If set, use real data, otherwise use asimov data")
-    parser.add_argument('-u', "--run", default='run2', help = "Which years to process", choices=["run2","run3"])
+    parser.add_argument('-u', "--run", default='run2', help = "Which years to process", choices=["all","run2","run3"])
```
Comment (kmohrman) [inline analysis/wwz/make_datacards.py:407]: Add your y22 and y23 stuff?

File: analysis/wwz/yld_dicts_for_comp.py [hunk 7]
```diff
@@ -131,6 +131,4 @@
...
...7429991, None], 'sr_4l_bdt_of_5': [1.3820360252848263, None], 'sr_4l_bdt_of_6': [0.6370829517071711, None], 'sr_4l_bdt_of_7': [0.7373091063982443, None], 'sr_4l_bdt_of_8': [0.14224750213284715, None], 'sr_4l_bdt_sf_coarse_1': [0.7009237124893868, None], 'sr_4l_bdt_sf_coarse_2': [0.16267797246784924, None], 'sr_4l_bdt_sf_coarse_3': [0.6321848204373318, None], 'sr_4l_bdt_sf_coarse_4': [0.10184534983170086, None], 'sr_4l_bdt_of_coarse_1': [0.9528041471262126, None], 'sr_4l_bdt_of_coarse_2': [0.7264357894264413, None], 'sr_4l_bdt_of_coarse_3': [1.4884176665941733, None], 'sr_4l_bdt_of_coarse_4': [0.31696312135532484, None], 'cr_4l_btag_of': [0.16857469943895115, None], 'cr_4l_btag_sf_offZ_met80': [0.10880440573578155, None], 'cr_4l_sf': [0.011165287596054951, None], 'sr_sf_all_cutbased': [0.8762679684966302, None], 'sr_of_all_cutbased': [1.6166507522632885, None], 'sr_all_cutbased': [1.8388597574058407, None], 'sr_sf_all_bdt': [1.1049485424016356, None], 'sr_of_all_bdt': [2.1290775787654397, None], 'sr_all_bdt': [2.398725206804065, None], 'sr_sf_all_bdtCoarse': [0.9632176783727954, None], 'sr_of_all_bdtCoarse': [1.936852464559312, None], 'sr_all_bdtCoarse': [2.1631425670536277, None]}}
```
Comment (kmohrman) [inline analysis/wwz/yld_dicts_for_comp.py:134]: Is this just 22, or the scaled 22+23 numbers?

File: analysis/wwz/make_datacards.py (hunk 5 above)
Comment (MatthewDittrich) [reply analysis/wwz/make_datacards.py:286]: Yes, but this is what we had before. Would we want to change this?

File: analysis/wwz/make_datacards.py (hunk 6 above)
Comment (MatthewDittrich) [reply analysis/wwz/make_datacards.py]: Done

File: analysis/wwz/make_datacards.py (hunk 4 above)
Comment (MatthewDittrich) [reply analysis/wwz/make_datacards.py:250]: Done

File: analysis/wwz/make_datacards.py [hunk 8]
```diff
@@ -220,6 +236,22 @@ def get_rate_systs(proc_lst):
 
     return out_dict
 
+def determine_updo_same(nom,up,down):
```
[the remark this answers is not in the document]
Comment (MatthewDittrich) [reply analysis/wwz/make_datacards.py:240]: Done

File: analysis/wwz/make_datacards.py (hunk 3 above)
Comment (MatthewDittrich) [reply analysis/wwz/make_datacards.py]: Done

File: analysis/wwz/get_wwz_yields.py (hunk 2 above)
Comment (MatthewDittrich) [reply analysis/wwz/get_wwz_yields.py]: Done

File: analysis/wwz/get_wwz_yields.py (hunk 1 above)
Comment (MatthewDittrich) [reply analysis/wwz/get_wwz_yields.py]: Done

File: analysis/wwz/yld_dicts_for_comp.py (hunk 7 above)
Comment (MatthewDittrich) [reply analysis/wwz/yld_dicts_for_comp.py]: I think I may have had this as a scaled 2022, but just 2022 would be better (updated)

File: analysis/wwz/make_datacards.py (hunk 5 above)
Comment (kmohrman) [reply analysis/wwz/make_datacards.py:288]: Let's revisit later. See https://github.com/cmstas/ewkcoffea/issues/45

**Comment by kmohrman** [comment]
Note for future reference: The new ref (for 22 alone, coarse BDT binning, with systematics, with simultaneous SR+CR fit): `1.8948`

**Pull request merged**